crates/python/Cargo.toml

```diff
@@ -26,6 +26,7 @@ tokio = "1.21"
 qcs-api = "0.2.1"
 rigetti-pyo3 = { version = "0.1.0-rc.0", features = ["extension-module", "complex"] }
 serde_json = "1.0.86"
+numpy = "0.17"
```

Comment:
note to self, `numpy` re-exports `ndarray` which is also needed by the `lib` crate. `numpy`, `pyo3`, and `pyo3-asyncio` all need to be version matched as well, so we should probably lift them into workspace dependencies with a comment about how to keep them in sync

Reply:
done